receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,242 @@
+		for _, org := range orgs {
+			projects, err := s.client.Projects(ctx, org.ID)
+			if err != nil {
+				s.log.Debug("error retrieving projects: %w")
+			}
+
+			// filter out any projects not specified in the config
+			filteredProjects := make(map[string]*mongodbatlas.Project)
+			for _, project := range projects {
+				if _, ok := cfgProjects[project.Name]; ok {
+					filteredProjects[project.Name] = project
+				}
+			}
+
+			// get clusters for each of the projects
+			for _, project := range filteredProjects {
+				resource := resourceInfo{Org: org, Project: project}
+				include, exclude := cfgProjects[project.Name].IncludeClusters, cfgProjects[project.Name].ExcludeClusters
+				clusters, err := s.client.GetClusters(ctx, project.ID, include, exclude)
+
+				// check to include or exclude clusters
+				switch {
+				//keep all clusters if include and exclude are not specified
+				case len(include) == 0 && len(exclude) == 0:
+					break
+				// include is initialized
+				case len(include) > 0 && len(exclude) == 0:
+					clusters, err = filterClusters(clusters, createStringMap(include), true)
+					break
+				// exclude is initialized
+				case len(exclude) > 0 && len(include) == 0:
+					clusters, err = filterClusters(clusters, createStringMap(exclude), false)
+					break
+				// both are initialized
+				default:
+					clusters = nil
+					s.log.Debug("Error can not have both include and exclude parameters initialized")
+				}
+
+				if err != nil {
+					s.log.Debug("error retreiving clusters from project")
+				}
+
+				// collection interval loop,
+				select {
+				case <-ctx.Done():
+					return
+				case <-stopper:
+					return
+				case <-time.After(collection_interval):
+					err = s.collectClusterLogs(clusters, cfgProjects, resource)
+					if err != nil {
+						s.log.Debug("Failure to collect logs from cluster")
+					}
+				}
+			}
+		}
+	}
+}
+
+func (s *receiver) collectClusterLogs(clusters []mongodbatlas.Cluster, cfgProjects map[string]*Project, r resourceInfo) error {
+	for _, cluster := range clusters {
+		hostnames := FilterHostName(cluster.ConnectionStrings.Standard)
+		for _, hostname := range hostnames {
+			r.Cluster = &cluster
+			r.Hostname = hostname
+			s.sendLogs(r, "mongodb.gz")
+			s.sendLogs(r, "mongos.gz")
+
+			if cfgProjects[r.Project.Name].EnableAuditLogs {
+				s.sendAuditLogs(r, "mongodb-audit-log.gz")
+				s.sendAuditLogs(r, "mongos-audit-log.gz")
+			}
+		}
+	}
+
+	return nil
+
+}
+
+func filterClusters(clusters []mongodbatlas.Cluster, keys map[string]string, include bool) ([]mongodbatlas.Cluster, error) {
+	var filtered []mongodbatlas.Cluster
+	for _, cluster := range clusters {
+		if _, ok := keys[cluster.ID]; (!ok && !include) || (ok && include) {
+			filtered = append(filtered, cluster)
+		}
+	}
+	return filtered, nil
+}
+
+func createStringMap(in []string) map[string]string {
+	list := make(map[string]string)
+	for i := range in {
+		list[in[i]] = in[i]
+	}
+
+	return list
+}
```

Comment:
This function stands out as a little funky to me; I think this is creating a set of the strings? If so, it would be clearer if this function was called `createStringSet` and the return type was `map[string]struct{}`.